src/main/java/org/symphonyoss/symphony/messageml/elements/DateSelector.java

```diff
@@ -25,17 +33,50 @@ public DateSelector(Element parent) {
   @Override
   public void validate() throws InvalidInputException {
     super.validate();
-    assertNoAttributes();
+    
+    if (getAttribute(NAME_ATTR) == null) {
+      throw new InvalidInputException("The attribute \"name\" is required");
+    }
+
     assertNoContent();
   }
 
   @Override
   public void asPresentationML(XmlPrintStream out) {
-    out.printElement(PRESENTATIONML_TAG, null, CLASS_ATTR, MESSAGEML_TAG);
+    Map<String, String> presentationAttrs = buildDateSelectorInputAttributes();
+    out.printElement(PRESENTATIONML_TAG, null, presentationAttrs);
```

Comment:
I think there's another method you don't have to pass this "null" in the middle.